File: Palindromer/Palindromer.java
import java.util.Scanner;

public class Palindromer {
  static boolean isPalindrome(String myWord, int lower, int upper) {
    boolean isPal = true;
    // if the start and end characters are unequal, the string is not a palindrome
    if(myWord.charAt(lower) != myWord.charAt(upper)) isPal = false;
    lower++;  // iterate and decrement position counters
    upper--;
    // if the string still may be a palindrome and the position counters have not intersected, call isPalindrome recursively
    if(isPal && upper >= lower) isPalindrome(myWord, lower, upper);
    return isPal;
  }

  public static void main(String[] args) {
    String myWord;
    Scanner myScanner = new Scanner(System.in);
    System.out.println("Enter a word and to be checked for palindromicity. Return an empty line to exit.");

    while(true) {
      System.out.print("Enter word: "); // prompt user for input
      myWord = myScanner.nextLine();
      if(myWord.equals("")) break;      // break if user returns empty line
      // call isPalindrome function with 0 and word length for args, declare palindromicity based on function output
      if(isPalindrome(myWord.toLowerCase(), 0, myWord.length() - 1)) System.out.println(myWord + " is a palindrome!");
      else System.out.println(myWord + " is NOT a palindrome!");
    }
  }
}

File: Palindromer/palindromer.cpp
#include <iostream>
#include <string>
#include <stdio.h>
#include <ctype.h>

using namespace std;

bool isPalindrome(string str, int lower, int upper)
{
    bool isPal = true;
    if (tolower(str[lower]) != tolower(str[upper])) isPal = false;

    upper--;
    lower++;

    if (isPal && (upper >= lower)) isPal = isPalindrome(str, lower, upper);
    return(isPal);
}

int main()
{
    string myWord;

    //user prompt
    cout << "Enter a word in lowercase and I will determine if it is a palindrome" << endl;
    cout << "Return an empty line to exit\n" << endl;

    //default values for lower and upper are beginning/end of string
    int upper = 0;
    int lower = 0;

    //loop to poll user and analyze input
    while(1)
    {
        //take user input
        cout << "Enter word: ";
        getline(cin, myWord);
        upper = myWord.length() - 1; //upper and lower default to the beginning and end of the user input

        //if the line is empty, break
        if(myWord.empty()) break;

        //call function to determine if palindrome
        if (isPalindrome(myWord, lower, upper)) cout << myWord << " IS a palindrome" << endl << endl;
        else cout << myWord << " is NOT a palindrome" << endl << endl;
    }

    return 0;
}
